### ISSUE TITLE
group.parent.filepath() returns garbage string

### ISSUE THREAD

**Issue by AgilentGCMS** [Unidata/netcdf4-python#377]
I am trying to get the file name from a group object of a netcdf-4 file. In a previous version of python-netcdf (1.0.7), this used to work (I still have it on another machine, which is how I can check):

```python
In [1]: fid = Dataset('point/point_input.nc4','r')

In [2]: gid = fid.groups['CO2_ff']

In [3]: gid.parent.filepath()
Out[3]: u'point/point_input.nc4'

In [4]: print gid.parent.filepath()
point/point_input.nc4
```

However, I recently updated to python-netcdf 1.1.7, and now this method does not work any more:

```python
In [1]: fid = Dataset('point/point_input.nc4','r')

In [2]: gid = fid.groups['CO2_ff']

In [3]: gid.parent.filepath()
Out[3]: u'p\x00o\x00i\x00n\x00t\x00/\x00p\x00o\x00i\x00n\x00t'

In [4]: print gid.parent.filepath()
point/point
```

So the unicode version has these null characters \x00 before each character, and the name is truncated at the first underscore. Is there another way of getting the file name from a group id?

Thanks,
Sourish Basu

**Comment by jswhit** [comment]
Sourish - I don't see that here.

    >>> from netCDF4 import Dataset
    >>> nc = Dataset('test.nc')
    >>> g = nc.groups['forecasts']
    >>> g.filepath()
    u'test.nc'
    >>> g.parent.filepath()
    u'test.nc'
    >>>

Could you post your file somewhere?

And no, there is no other way to get the filename from a group or dataset id.

**Comment by jswhit** [comment]
It may have more to do with the netcdf library version that was linked than the version of the python module.

Try this on the working and non-working installations:

    >>> import netCDF4
    >>> netCDF4.__netcdf4libversion__
    u'4.3.3-rc2'

**Comment by AgilentGCMS** [comment]
Jeff, you are probably right. On the non-working system I installed python-netcdf-1.0.7, which was the version on the working system, and the problem persisted. So on the non-working system I went back to the latest, 1.1.7. Here are the netcdf library versions:

On the working system:
```python
In [4]: import netCDF4

In [5]: netCDF4.__netcdf4libversion__
Out[5]: u'4.3.0'
```

On the non-working system:
```python
In [5]: import netCDF4

In [6]: netCDF4.__netcdf4libversion__
Out[6]: u'4\x00.\x003\x00.\x001\x00.\x001\x00'

In [7]: print netCDF4.__netcdf4libversion__
4.3.1.1
```

In the original report, I said that on the non-working system, I had the problem that (a) characters were preceded by the null character \x00, and (b) the filepath got truncated. It looks like problem (a) is not unique to filepath; other strings in the netCDF4 library have that issue too.
